docs/advanced-topics/wes/wes-agc-tutorial.rst

```diff
@@ -0,0 +1,329 @@
+WES tutorial for AWS AGC
+========================
+Amazon Web Services' (AWS) Amazon Genomics CLI (AGC) is a command line tool for launching cloud infrastructure
+within AWS accounts that can be used to execute genomics workflows. The infrastructure deployed by AGC implements the WES
+standard, and thus can be directly communicated with by using the Dockstore CLI.
+
+Check out the official AGC `GitHub page <https://github.com/aws/amazon-genomics-cli>`_.
+
+Download and install AWS AGC
+----------------------------
+AGC provides a `quick-start guide <https://aws.github.io/amazon-genomics-cli/docs/getting-started/>`_ for initial setup
+and getting familiar with the tool. The following workflow execution tutorials will cover all steps for using AGC once
+it has been `installed <https://aws.github.io/amazon-genomics-cli/docs/getting-started/installation/>`_.
+
+
+
+
+Tutorial topics
+----------------
+The following WDL workflow tutorials will covering deploying the proper infrastructure using AGC.
+
+1. Deploy an AGC project and context
+2. Configure the Dockstore CLI to communicate with AGC infrastructure
+3. Launching a workflow
+
+.. note::
+
+    1. AWS AGC requires a special JSON file to passed as the test parameter file for workflows using the format
+
+        .. code:: text
+
+            {
+                "workflowInputs": <pathToInputJSON>
+            }
+
+    2. AWS AGC references descriptor files by URLs. The Dockstore CLI communicates to AGC infrastructure using TRS URLs,
+    which require the referenced workflow to be publicly published on Dockstore.
+
+Configuring AGC and the Dockstore CLI
+----------------------------------------
+1. Create a file named ``agc-project.yaml`` that contains:
+
+    .. code:: text
+
+        name: dockstoreAgcTutorialProject
+        data:
+          - location: s3://human-pangenomics
+            readOnly: true
+        schemaVersion: 1
+        contexts:
+          ctx1:
+            engines:
+              - type: wdl
+                engine: cromwell
+
+This will create an AGC project named ``dockstoreAgcTutorialProject``, with a single context named ``ctx1``. The section containing
+
+    .. code:: text
+
+        data:
+          - location: s3://human-pangenomics
+            readOnly: true
+
+Configures the AGC contexts in this project to have read access to the AWS S3 bucket ``human-pangenomics``. This bucket is an open-access data
```

Comment:
I'm a little confused by this one as I don't know it, so comments questions:

1. Does this actual give access to the bucket, e.g., modify IAM permissions? I'd be surprised, especially if the bucket is in a different account.
2. If I don't do this, and it's an open access bucket, will things not work?

Reply:
This is an internal AGC specification. The AGC infrastructure will refuse to communicate with an S3 bucket that isn't in the configuration at deploy time. 

From what I understand based on their setup tutorials, if your account has access to an S3 bucket **and** the AGC project allows read/write access to the bucket, then you can access the bucket's files. So, a bucket without open access in a different account will fail to be reached, but any public S3 bucket or a given S3 bucket in your own account should work (barring bucket level access policies).